tensorflow/compiler/mlir/tools/kernel_gen/transforms/passes.td

```diff
@@ -54,6 +54,31 @@ def TFToJITInvocationPass : FunctionPass<"tf-to-jit-invocation"> {
              /*default=*/"", "Max rank that this kernel supports">,
       Option<"enable_ftz_", "enable-ftz", "bool", /*default=*/"",
              "Enable the denormal flush to zero mode when generating code">,
+      Option<"index_64bit_", "index_64bit", "bool", /*default=*/"",
+             "Enable the 64 bit indexing for GPU kernels">,
+      Option<"cpu_codegen_", "cpu-codegen", "bool", /*default=*/"",
+             "CPU codegen (false implies GPU)">,
+  ];
+}
+
+def IndexSizerPass : FunctionPass<"tf-64bit-indexer-invocation"> {
+  let summary = "Pass to rewrite all TF operations to JIT invocations";
```

Comment:
Can you mention that the pass creates dispatch code and that it creates the JIT invocation only in case the argument is too large, otherwise?

Reply:
description added.